modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/KeyValueStorageUtils.java

```diff
@@ -43,14 +43,20 @@ public static int indexToCompact(long[] keyRevisions, long compactionRevisionInc
         int i = binarySearch(keyRevisions, compactionRevisionInclusive);
 
         if (i < 0) {
-            if (i == -1) {
+            i = ~i;
+
+            if (i == 0) {
                 return NOT_FOUND;
             }
 
-            i = -(i + 2);
+            i--;
```

Comment:
For the reviewer:

new code in this block is doing exactly the same thing. I just think it is simpler than it was before. `-(i+2)` is impossible to comprehend

Reply:
Maybe the code can be simplified if we would do `int i = binarySearch(keyRevisions, compactionRevisionInclusive + 1);` instead. I didn't think about it yesterday